### ISSUE TITLE
[QUESTION] Use another model and table for roles and permissions

### ISSUE THREAD

**Issue by denitsa-cm** [spatie/laravel-permission#633]
First, thank you for this package - it's great! 

Then, I have a question that is project-specificI which I would love your advice on.

In our project, we have a `users` and `accounts` tables and models because a user can have many accounts. I would like to assign roles and permissions to an `account` and not a `user`. Saving the relations on the Account works. My question is how to go about retrieving the relations from the role/permission side. 

I would like to have something like `$role->accounts()`. Would it be okay to have an `accounts()` function in an extended `Role` model or would you recommend something else? 

Thanks in advance! 

P.S. New to Laravel as well. :)

**Comment by drbyte** [comment]
At first glance, yes, I think extending things is probably what's needed to adapt it to your project's unique needs.

**Comment by kreierson** [comment]
Hey @denitsa-cm 

I'm curious to know what your setup looks like. I'm looking to implement this exact same scenario.
Originally I set up two guards, one for users (web guard) and another for accounts (account guard).

Then when the user logged in through the web guard, if they had multiple accounts I presented them with the available accounts. When they clicked the account they wanted to log into, I logged them out of the user web guard, then logged them into the account guard, which uses the model that stores all their permissions for that account. I works pretty well but just curious how you went about it. I'm wondering if getting away from multiple guards would make it a little easier, but not sure how to implement.
